### PR TITLE
Spot information page

### PR DESCRIPTION
**Pull request by LuizGuerra**
Spot information view for the driveway owner
* Changed double column to box + [column column] to be able to scroll the original content
* Added all missing fields
* New _widget_ for showing when the spot is available during the week

Screenshot 2024-10-31 at 9 31 25 PM

_No integration with previously added data (everything from before was mocked too) and the url image library crashes my computer_

### CODE REVIEW

**Comment by alyssa-blair** [comment]
Will probably need to merge this after the demo, there's also merge conflicts probably due to adding the delete button on this page. I'll take a look at it once the conflicts are resolved.

**Comment by LuizGuerra** [comment]
> Will probably need to merge this after the demo, there's also merge conflicts probably due to adding the delete button on this page. I'll take a look at it once the conflicts are resolved.

Yeah, I saw that, I can do a rebase of that one pretty easily after the delete feature gets merged

**Comment by alyssa-blair** [comment]
> Member

The delete feature is already merged

**Comment by LuizGuerra** [comment]
Oh shoot, I saw it was open yesterday so went to review it – but didn't reply to it since it was 24h before the presentation

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:325]: this box with the images should be: 
               Box(
                    modifier = Modifier
                        .width(200.dp).height(154.dp)
                        .padding(horizontal = 4.dp)
                        .background(Color.LightGray),
                ) {
                    Image(
                        painter = painterResource(Res.drawable.image_placeholder),
                        contentDescription = null,
                        contentScale = ContentScale.FillBounds,
                    )
                }
This does stretch the image if they are not the same dimension ratio so you could address that if you'd like by removing the grey behind, but probably not necessary for now.

You'll need to import:
import cruiser.composeapp.generated.resources.image_placeholder

File: composeApp/src/main/res/drawable/image_placeholder.webp
Comment (alyssa-blair) [inline composeApp/src/main/res/drawable/image_placeholder.webp:1]: You should move this to into commonMain/composableResources/drawable/ so this is accessible to the views in commonMain. If possible would be good to have as a png to match the format of our other images.

**Comment by alyssa-blair** [comment]
<vector xmlns:android="http://schemas.android.com/apk/res/android"
    android:width="28dp"
    android:height="26dp"
    android:viewportWidth="28"
    android:viewportHeight="26">
  <path
      android:pathData="M7.961,25.887C8.418,25.887 8.758,25.652 9.309,25.137L13.539,21.27H21.461C24.953,21.27 26.828,19.348 26.828,15.914V6.938C26.828,3.504 24.953,1.57 21.461,1.57H6.367C2.875,1.57 1,3.492 1,6.938V15.914C1,19.359 2.875,21.27 6.367,21.27H6.918V24.68C6.918,25.406 7.293,25.887 7.961,25.887ZM13.926,13.594C13.316,13.594 12.977,13.254 12.965,12.633L12.813,6.258C12.801,5.637 13.258,5.18 13.914,5.18C14.559,5.18 15.039,5.648 15.027,6.27L14.852,12.633C14.84,13.266 14.5,13.594 13.926,13.594ZM13.926,17.52C13.223,17.52 12.613,16.969 12.613,16.266C12.613,15.574 13.211,15.012 13.926,15.012C14.629,15.012 15.227,15.563 15.227,16.266C15.227,16.98 14.617,17.52 13.926,17.52Z"
      android:fillColor="#C90000"/>
</vector>
for the user_warning.xml file

**Comment by LuizGuerra** [comment]
Ready to merge

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:137]: should be $50/day

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:157]: Should probably use the arrow icon from Icons.AutoMirrored.Default.ArrowForward here and make this a button so it's consitent with other buttons. This will follow a similar format to the buttons on the profile view

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:361]: Might make sense to create a (String, Boolean) map here instead of using two separate lists

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:100]: Should maybe reduce this padding to 16.dp to make it more consistent with other pages

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:115]: Can remove this row and apply the modifier to the text box directly

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:133]: Can remove this row and apply the modifier to the text box directly

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:145]: Can remove this row and apply the modifier to the text box directly

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:154]: Can remove this row and apply the modifier to the text box directly

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:170]: Can remove this row and apply the modifier to the text box directly

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:212]: Reduce padding to 16.dp here as well

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:277]: Spacer here is further than the previous one, should probably make them the same size of separate them further

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:328]: remove unused parameter

File: composeApp/src/commonMain/composeResources/drawable/image_placeholder.png
Comment (alyssa-blair) [inline composeApp/src/commonMain/composeResources/drawable/image_placeholder.png:1]: don't think this is actually used anywhere

**Review by alyssa-blair — CHANGES_REQUESTED** [review]

**Comment by LuizGuerra** [comment]
On it

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:361]: Had to make something similar in this PR for an example of the map:  https://github.com/proj-overdrive/cruiser/pull/32/files#diff-a4be91e8e0b6d7e4bf65124dac7895672f573a7373b57fa620eba0105ebc6628R425

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (LuizGuerra) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:100]: This page had a double 16.dp before, that's why I was using 32 here. Just updated it (plus the other references)

File: composeApp/src/commonMain/composeResources/drawable/image_placeholder.png
Comment (LuizGuerra) [reply composeApp/src/commonMain/composeResources/drawable/image_placeholder.png:1]: Somehow the part where it was used reverted with git – updated it so it's now being used while we don't have the images.

**Comment by LuizGuerra** [comment]
Ready to review*

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt
Comment (alyssa-blair) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt:146]: Should be a weight spacer here so that the arrow sits at the end of the box

**Review by alyssa-blair — APPROVED** [review]
Just one nit, need to resolve merge conflicts though

**Comment by LuizGuerra** [comment]
Solved the merge conflict

**Pull request merged**
